protocol/contracts/libraries/Silo/LibDelegate.sol

```diff
@@ -0,0 +1,56 @@
+/**
+ * SPDX-License-Identifier: MIT
+ **/
+
+pragma solidity =0.7.6;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import "../../C.sol";
+import "../LibAppStorage.sol";
+
+/**
+ * @author Publius
+ * @title Lib Delegate
+ **/
+library LibDelegate {
+    using SafeMath for uint256;
+
+    /**
+     * Delegate
+     **/
+
+    function getAllowance(
+        address account,
+        bytes4 selector,
+        address spender
+    ) internal view returns (uint256 allowance) {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+        allowance = uint256(s.a[account].functionApprovals[selector][spender]);
+    }
+
+    function setAllowance(
+        address account,
+        bytes4 selector,
+        address delegatee,
+        bytes32 approval
+    ) internal {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+        s.a[account].functionApprovals[selector][delegatee] = approval;
+    }
+
+    function spendAllowance(
+        address account,
+        bytes4 selector,
+        address spender,
+        uint256 beans
```

Comment:
Instead of `beans` can we use the word `amount`? Not all delegate functions will be denominated in Beans.